client/post-editor/media-modal/image-editor/image-editor-crop.jsx

```diff
@@ -0,0 +1,224 @@
+		}
+
+		if ( newValues.hasOwnProperty( 'left' ) ) {
+			newValues.left = this.state.right - finalWidth;
+		} else if ( newValues.hasOwnProperty( 'right' ) ) {
+			newValues.right = this.state.left + finalWidth;
+		}
+
+		this.setState( newValues );
+	},
+
+	onTopLeftDrag( x, y ) {
+		this.updateCrop( {
+			top: y,
+			left: x
+		} );
+	},
+
+	onTopRightDrag( x, y ) {
+		this.updateCrop( {
+			top: y,
+			right: x
+		} );
+	},
+
+	onBottomRightDrag( x, y ) {
+		this.updateCrop( {
+			bottom: y,
+			right: x
+		} );
+	},
+
+	onBottomLeftDrag( x, y ) {
+		this.updateCrop( {
+			bottom: y,
+			left: x
+		} );
+	},
+
+	onDragStop() {
+		const currentTop = this.state.top - this.props.topBound,
+			currentLeft = this.state.left - this.props.leftBound,
+			currentWidth = this.state.right - this.state.left,
+			currentHeight = this.state.bottom - this.state.top,
+			imageWidth = this.props.rightBound - this.props.leftBound,
+			imageHeight = this.props.bottomBound - this.props.topBound;
+
+		this.props.imageEditorCrop(
+			currentTop / imageHeight,
+			currentLeft / imageWidth,
+			currentWidth / imageWidth,
+			currentHeight / imageHeight );
+	},
+
+	render() {
+		const { top, left, right, bottom } = this.state,
+			width = right - left,
+			height = bottom - top;
+
+		return (
+			<div>
+				<div
+					ref="border"
+					style={ { top: top + 'px', left: left + 'px', width: width + 'px', height: height + 'px' } }
+					className="editor-media-modal-image-editor__crop"></div>
+				<Draggable
+					onDrag={ this.onTopLeftDrag }
+					onStop={ this.onDragStop }
+					x={ left }
+					y={ top }
+					controlled
+					bounds={ { top: this.props.topBound, left: this.props.leftBound, bottom, right } }
+					ref="topLeft"
+					className="editor-media-modal-image-editor__crop-handle editor-media-modal-image-editor__crop-handle-nwse" />
+				<Draggable
+					onDrag={ this.onTopRightDrag }
+					onStop={ this.onDragStop }
+					y={ top }
+					x={ right }
+					controlled
+					bounds={ { top: this.props.topBound, right: this.props.rightBound, bottom, left } }
+					ref="topRight"
+					className="editor-media-modal-image-editor__crop-handle editor-media-modal-image-editor__crop-handle-nesw" />
+				<Draggable
+					onDrag={ this.onBottomRightDrag }
+					onStop={ this.onDragStop }
+					y={ bottom }
+					x={ right }
+					controlled
+					bounds={ { bottom: this.props.bottomBound, right: this.props.rightBound, top, left } }
+					ref="bottomRight"
+					className="editor-media-modal-image-editor__crop-handle editor-media-modal-image-editor__crop-handle-nwse" />
+				<Draggable
+					onDrag={ this.onBottomLeftDrag }
+					onStop={ this.onDragStop }
+					y={ bottom }
+					x={ left }
+					bounds={ { bottom: this.props.bottomBound, left: this.props.leftBound, top, right } }
+					ref="bottomLeft"
+					className="editor-media-modal-image-editor__crop-handle editor-media-modal-image-editor__crop-handle-nesw" />
```

Comment:
Optional, but the className prop is getting hard to read from the long class names. Maybe create a variable here, using the classnames module or native classlist if we have a shim.